linq4j/src/main/java/org/apache/calcite/linq4j/tree/MethodDeclaration.java

```diff
@@ -59,20 +61,38 @@ public MethodDeclaration(int modifier, String name, Type resultType,
   }
 
   @Override public void accept(ExpressionWriter writer) {
+    final ExpressionWriter tempWriter = writer.duplicateState();
     String modifiers = Modifier.toString(modifier);
-    writer.append(modifiers);
+    tempWriter.append(modifiers);
     if (!modifiers.isEmpty()) {
-      writer.append(' ');
+      tempWriter.append(' ');
     }
     //noinspection unchecked
-    writer
+    tempWriter
         .append(resultType)
         .append(' ')
         .append(name)
         .list("(", ", ", ")",
             () -> (Iterator) parameters.stream().map(ParameterExpression::declString).iterator())
         .append(' ')
         .append(body);
+
+    // From Flink TableConfigOptions.MAX_LENGTH_GENERATED_CODE
+    // Note that Flink uses 4000 rather than 64KB explicitly:
+    // "Specifies a threshold where generated code will be split into sub-function calls.
+    //  Java has a maximum method length of 64 KB. This setting allows for finer granularity if
+    //  necessary.
+    //  Default value is 4000 instead of 64KB as by default JIT refuses to work on methods with
+    //  more than 8K byte code."
+    final int flinkDefaultMaxGeneratedCodeLength = 4000;
+
+    // From Flink TableConfigOptions.MAX_MEMBERS_GENERATED_CODE
```

Comment:
these Flink considerations seem in appropriate in Calcite, unless we can make them generic so we do not reference Flink at all.

Reply:
I've removed references to Flink here.